### PR TITLE
feat: add new operational authority group

### PR DESCRIPTION
**Pull request by lumtis**
# Description

- Add a operational authority group for operational purpose (non-sensitive network param changes)
- Move all the admin messages to operational messages except the following which are sensitive as discussed in the doc:
  - `MsgUpdateTssAddress`
  - `MsgUpdateContractBytecode`
  - `MsgUpdateSystemContract`
  - `MsgMigrateTssFunds`

Change the migration script to set the current group 2 address to group admin and operational at the same time. The operational address should be changed with a gov prop later (with less required signatures)

Closes: https://github.com/zeta-chain/node/issues/1715

### CODE REVIEW

File: proto/authority/policies.proto [hunk 1]
```diff
@@ -8,8 +8,9 @@ option go_package = "github.com/zeta-chain/zetacore/x/authority/types";
 // PolicyType defines the type of policy
 enum PolicyType {
   option (gogoproto.goproto_enum_stringer) = true;
-  groupEmergency = 0;
-  groupAdmin = 1;
+  groupEmergency = 0; // Used for emergency situations that require immediate action
+  groupOperational = 1; // Used for operational tasks like changing non-sensitive protocol parameters
+  groupAdmin = 2; // Used for administrative tasks like changing sensitive protocol parameters or moving funds
```
Comment (kingpinXD) [inline proto/authority/policies.proto:13]: Maybe use `groupAdministrative` , it sounds more similar to the other two group names

File: x/fungible/keeper/msg_server_update_zrc20_paused_status.go [hunk 2]
```diff
@@ -31,7 +31,7 @@ func (k msgServer) UpdateZRC20PausedStatus(
 	// unpausing requires group2 admin
```
Comment (kingpinXD) [inline x/fungible/keeper/msg_server_update_zrc20_paused_status.go:31]:
```suggestion
	// unpausing requires operational policy
```

File: x/observer/keeper/msg_server_update_observer.go [hunk 3]
```diff
@@ -81,7 +81,7 @@ func (k Keeper) CheckUpdateReason(ctx sdk.Context, msg *types.MsgUpdateObserver)
 	case types.ObserverUpdateReason_AdminUpdate:
 		{
 			// Group admin is authorized to update observer
```
Comment (kingpinXD) [inline x/observer/keeper/msg_server_update_observer.go:83]:
```suggestion
			// Operational policy is required to update an observer for admin update 
```

Comment (kingpinXD) [reply x/observer/keeper/msg_server_update_observer.go:83]: Alos I think this should be policyAdmin , this affects the observer set , and we would almost never need to do this . Observers if tombstoned have the option to select `ObserverUpdateReason_Tombstoned` reason

File: x/observer/migrations/v7/migrate.go [hunk 4]
```diff
@@ -29,16 +29,24 @@ func MigratePolicies(ctx sdk.Context, observerKeeper observerKeeper) error {
 	for _, adminPolicy := range params.AdminPolicy {
 		if adminPolicy != nil {
 
-			// Convert group1 -> emergency and group2 -> admin
-			policyType := authoritytypes.PolicyType_groupAdmin
 			if adminPolicy.PolicyType == types.Policy_Type_group1 {
-				policyType = authoritytypes.PolicyType_groupEmergency
+				// for policy group 1, we set the policy type to emergency
```
Comment (kingpinXD) [inline x/observer/migrations/v7/migrate.go:33]: So even after this mightation observers.AdminPolicy would exist . Would that get removed in the next release?

Comment (lumtis) [reply x/observer/migrations/v7/migrate.go:33]: Yes, we are removing all params in any case for v47 upgrade

File: x/crosschain/keeper/msg_server_whitelist_erc20.go [hunk 5]
```diff
@@ -28,7 +28,7 @@ func (k msgServer) WhitelistERC20(goCtx context.Context, msg *types.MsgWhitelist
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
 	// check if authorized
-	if !k.GetAuthorityKeeper().IsAuthorized(ctx, msg.Creator, authoritytypes.PolicyType_groupAdmin) {
+	if !k.GetAuthorityKeeper().IsAuthorized(ctx, msg.Creator, authoritytypes.PolicyType_groupOperational) {
```
[the remark this answers is not in the document]
Comment (lumtis) [reply x/crosschain/keeper/msg_server_whitelist_erc20.go:31]: Yes [255fbf0](https://github.com/zeta-chain/node/pull/1935/commits/255fbf0f3973bae1231de8a071785d508913d8df)

[the remark this answers is not in the document]
Comment (lumtis) [reply x/crosschain/keeper/msg_server_whitelist_erc20.go:31]: Keeping the name `IsAuthorized` in `observer` to check observer was also a bit unclear, renamed it to `IsNonTombstonedObserver`: [8ac6e7b](https://github.com/zeta-chain/node/pull/1935/commits/8ac6e7ba4bba405cd0da9995eb88e25b99a58a7f)

File: proto/authority/policies.proto (hunk 1 above)
Comment (lumtis) [reply proto/authority/policies.proto:13]: I would prefer to keep it short personally

File: x/observer/keeper/msg_server_update_observer.go (hunk 3 above)
Comment (lumtis) [reply x/observer/keeper/msg_server_update_observer.go]: Sounds good to use group admin [ba9f135](https://github.com/zeta-chain/node/pull/1935/commits/ba9f13515af948497c882c29ea761be17bacecd9) if sensitive and not frequently used

File: app/ante/ante.go [hunk 6]
```diff
@@ -102,7 +102,7 @@ func NewAnteHandler(options HandlerOptions) (sdk.AnteHandler, error) {
 			// if tx is a system tx, and singer is authorized, use system tx handler
 
 			isAuthorized := func(creator string) bool {
```
[the remark this answers is not in the document]
Comment (lumtis) [reply app/ante/ante.go:104]: I think it's fine to keep it more general here

**Review by kingpinXD — APPROVED** [review]

**Pull request merged**